learning/step_by_step/godot_design_philosophy.rst

```diff
@@ -0,0 +1,145 @@
+Godot’s design philosophy
+=========================
+
+**Every game engine is different and fits different needs**. They do not
+only offer a wide range features: each engine’s design is unique. This
+leads to different workflows and different ways to reason about your
+games’ structure. This all stems from their design philosophy.
+
+This page is here to **help you understand how Godot works**, starting
+from some of its core pillars. It is not a list of available features or
+an engine comparison. To know whether any engine can be a good fit for
+your project or not, you need to both try it out for yourself and
+understand its design and limitations.
+
+Object-oriented design
+----------------------
+
+Godot embraces object-oriented design at its core with its **flexible
+scene system and Node hierarchy**. It tries to stay away from strict
+programming patterns to offer an intuitive way to structure your game.
+
+A Godot scene could be a Weapon, a Character, an Item, a Door, a Level,
+part of a level… anything you’d like. It works like a class in pure code
+except you’re free to design your scene using the editor, using only the
+code, or mixing and matching the two.
+
```

Comment:
IMO, before diving into inheritance (which is powerful but also has more *baggage*), it would be worth it to drop a sentence about *composability* here:

You can create a light, then create a lantern that uses that light and then create a city scene that uses that same lantern and change all of those in their own scenes and see the changes reflected anywhere. ***Nested prefabs**, we got them.* :P 

Composition over inheritance and all that jazz.